### PR TITLE
Articles/testing/refs

### PR DESCRIPTION
**Pull request by pytlesk4**
Scenario Refs article

### CODE REVIEW

**Comment by rossmcdonald** [comment]
@pytlesk4 I think you branched off of the same branch in https://github.com/stoplightio/docs/pull/132? `pass-data-steps` and `scenarios-introduction` should be removed.

File: articles/testing/ref-other-scenarios.md [hunk 1]
```diff
@@ -1 +1,29 @@
+# Scenario References
 
+References are a special type of step that allow you to include other scenarios in your scenario. This is a very powerful concept that can dramatically reduce the duplication in your scenarios.
```
Comment (rowa97) [inline articles/testing/ref-other-scenarios.md]: we may want to consider reframing the first line, it's a bit confusing with the repetition,

File: articles/testing/ref-other-scenarios.md [hunk 2]
```diff
@@ -1 +1,29 @@
...
+
+To use them, add a step and change it's type from "HTTP/Script" to "REF". This will expose a dropdown that lists all of the scenarios in your collection. Select a value from the dropdown to create the ref to that scenario.
+
+In the screenshot below, [Todo: Utility CRUD](https://next.stoplight.io/stoplight/hub/blob/master/tests.scenarios.yml?edit=%23%2Fscenarios%2Ftodo-utility) and [User: Utility CRUD](https://next.stoplight.io/stoplight/hub/blob/master/tests.scenarios.yml?edit=%23%2Fscenarios%2Fuser-utility) scenario have one step that refrences [Entity CRUD](https://next.stoplight.io/stoplight/hub/blob/master/tests.scenarios.yml?edit=%23%2Futilities%2Fcrud) from Utilities. Entity CRUD does all the heavy lifting: it CREATES, GETS, UPDATES, DELETES, and repeats GETS (To verify entity was deleted) an entity.
```
Comment (rowa97) [inline articles/testing/ref-other-scenarios.md]: is caps appropriate here (bottom of the paragraph)?

File: articles/testing/ref-other-scenarios.md [hunk 3]
```diff
@@ -1 +1,29 @@
+# Scenario References
 
+References are a special type of step that allow you to include other scenarios in your scenario. This is a very powerful concept that can dramatically reduce the duplication in your scenarios.
+
+To reference another scenario, add a step to an existing scenario and change it's type from "HTTP/Script" to "REF". This will expose a dropdown that lists all of the scenarios in your collection. Select a value from the dropdown to create the ref to that scenario.
```
Comment (rowa97) [inline articles/testing/ref-other-scenarios.md]: break out into steps with a # How header, most basic necessary steps for referencing another scenario

File: articles/testing/ref-other-scenarios.md (hunk 2 above)
Comment (rossmcdonald) [reply articles/testing/ref-other-scenarios.md]: I think it should be switched to lower-case. We should also use the CRUD terminology, so instead of 'gets' use 'reads'.

File: articles/testing/ref-other-scenarios.md (hunk 1 above)
Comment (rossmcdonald) [reply articles/testing/ref-other-scenarios.md]: Agreed on the first sentence being a little confusing (I heard you like scenarios in your scenario).

What about something like this:

> References in Stoplight allow you to re-use components of your existing projects and specifications without having to recreate or duplicate work. Within scenarios, references can be used to include other scenarios as a step type, allowing for common scenarios to be re-used multiple times. By leveraging scenario references, tests can be chained together with ease, allowing users to create more complex workflows without the extra complexity or effort.

File: articles/testing/ref-other-scenarios.md [hunk 4]
```diff
@@ -1 +1,29 @@
...
+
+In the screenshot below, [Todo: Utility CRUD](https://next.stoplight.io/stoplight/hub/blob/master/tests.scenarios.yml?edit=%23%2Fscenarios%2Ftodo-utility) and [User: Utility CRUD](https://next.stoplight.io/stoplight/hub/blob/master/tests.scenarios.yml?edit=%23%2Fscenarios%2Fuser-utility) scenario have one step that refrences [Entity CRUD](https://next.stoplight.io/stoplight/hub/blob/master/tests.scenarios.yml?edit=%23%2Futilities%2Fcrud) from Utilities. Entity CRUD does all the heavy lifting: it CREATES, GETS, UPDATES, DELETES, and repeats GETS (To verify entity was deleted) an entity.
+
+$.ctx values saved in ref steps **ARE** available in subsequent steps. The ref step in the screenshot below saves a $.ctx.accessToken property (not displayed), that the "Create Org" step then uses.
```
Comment (rossmcdonald) [inline articles/testing/ref-other-scenarios.md]: Maybe rephrase this to avoid using `$.ctx` to refer to the scenario context, instead using the name 'scenario context'. I also think we should avoid bolded, all-caps words. Maybe rephrase this section to:

> Values stored in the scenario context (using the `$.ctx` syntax) during referenced steps **are** available in subsequent steps. The referenced step in the screenshot below saves a `$.ctx.accessToken` property (not displayed), that the "Create Org" step then uses.

File: articles/testing/ref-other-scenarios.md [hunk 5]
```diff
@@ -1 +1,29 @@
...
+
+![](https://s3.amazonaws.com/cdn.stoplight.io/help-portal/scenarios/create+org+%2B+user+with+ref.png)
+_need to update image_
+
+### Viewing Reference Output
+
+When a reference is run as part of a scenario, its output will expand so that you can view the results of each step. In the screenshot below, we are looking at the results of the `Todo: Utility CRUD` scenario pictured above.
+
+Tests and assertions made in a referenced scenario **will** contribute to the overall pass/fail count of the calling scenario.
```
Comment (rossmcdonald) [inline articles/testing/ref-other-scenarios.md]: Maybe make this a callout.
